contrib/datawave-quickstart/bin/services/accumulo/bootstrap.sh

```diff
@@ -117,7 +117,61 @@ function accumuloStop() {
 }
 
 function accumuloStatus() {
-    accumuloIsRunning && echo "Accumulo is running. PIDs: ${DW_ACCUMULO_PID_LIST} ${DW_ZOOKEEPER_PID_LIST}" || echo "Accumulo is not running"
+    # define vars for accumulo processes
+    local _gc
+    local _master
+    local _monitor
+    local _tracer
+    local _tserver
+
+    echo "======  Accumulo Status  ======"
+    local _opt=pid
+    local _arg
+
+    for _arg in $(jps -lm | grep org.apache.accumulo.start); do
+        case ${_opt} in
+            pid)
+                _pid=${_arg}
+                _opt=class;;
+            class) _opt=component;;
+            component)
+                local _none
+                case "${_arg}" in
+                    gc) _gc=${_pid};;
+                    master) _master=${_pid};;
+                    monitor) _monitor=${_pid};;
+                    tracer) _tracer=${_pid};;
+                    tserver) _tserver=${_pid};;
+                    *) _none=true;;
+                esac
+
+                test -z "${_none}" && info "${_arg} => ${_pid}"
+                _opt=address
+                unset _none
+                _pid=;;
+            address) _opt=addrValue;;
+            addrValue) _opt=pid;;
+        esac
+    done
+
+    test -z "${_gc}" && error "accumulo gc is not running"
+    test -z "${_master}" && error "accumulo master is not running"
+    test -z "${_monitor}" && error "accumulo monitor is not running"
+    test -z "${_tracer}" && error "accumulo tracer is not running"
+    test -z "${_tserver}" && error "accumulo tserver is not running"
+
+    unset _pid
+    local -r _zoo=$(jps -l | grep org.apache.zookeeper)
+    test -n "${_zoo}" && {
+        set -- ${_zoo}
```

Comment:
Overwriting the function args could lead to bugs down the road if someone wants to add to this function and puts something at the end assuming they can work with the args. Everything in lines 163-168 could be replaced with
```bash
_pid=$(jps -l | grep org.apache.zookeeper | cut -d' ' -f 1)
```

Reply:
There are typically several alternative solutions. This may be a style issue but I don't agree with your assessment that this could lead to bugs. I prefer to use internal bash feature whenever possible rather than creating a new process.

Reply:
> I don't agree with your assessment that this could lead to bugs

`set -- ${_zoo}` replaces the function arguments array `$@` with the value of the `_zoo` variable. If someone doesn't understand that and went to use the args later on in some addition to this function, they'd get something they didn't expect. Or worse, if they tested without zookeeper running, then everything would work as they expect (since the args wouldn't be overwritten) and you wouldn't find out until later that the change was bad. Now since it doesn't seem any args are passed, it probably doesn't matter in this case.

Personally, I find:
```bash
_pid=$(jps -l | grep org.apache.zookeeper | cut -d' ' -f 1)
```
a lot easier to read and understand vs
```bash
    unset _pid
    local -r _zoo=$(jps -l | grep org.apache.zookeeper)
    test -n "${_zoo}" && {
        set -- ${_zoo}
        _pid=$1
    }
```
at the expense of an extra pipe through `cut` (you've already created the sub-shell and two processes).

Anyway, since args aren't passed to this function, I'll retract my objection unless someone else also has an issue with this.

Reply:
Personally I prefer the readability of what @brianloss posted. Also, there isn't a large enough gain to justify the difference in readability, but that's just my opinion.